### ISSUE TITLE
Import pybind11_tests failed

### ISSUE THREAD

**Issue by LovelyBuggies** [scikit-hep/boost-histogram#312]
I successfully create my python3 test environment `.env`, which is able to run jupyter lab. But when I tried to test using pytest method according to `python3 -m pytest`, an error was thrown like this:

``` bash
Test session starts (platform: darwin, Python 3.7.0, pytest 5.3.5, pytest-sugar 0.9.2)
benchmark: 3.2.3 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /Users/nino/Documents/GitHub/boost-histogram
plugins: sugar-0.9.2, benchmark-3.2.3
collecting ... Failed to import pybind11_tests from pytest:
  ModuleNotFoundError: No module named 'pybind11_tests'
```

Why this happens? How to deal with it? 

*P.S. I have my pytest packages (`pytest`, `pytest-sugar`, and `pytest-benchmark`) all installed in my `.env` environment and I run pytest in `.env`, too.*

**Comment by henryiii** [comment]
You are doing a great job of finding my mistakes. :) I guess I habitually run `python3 -m pytest tests`, which forces the `tests` dir to be the only place searched. Without that, we aren't limiting the search locations, so it picks up `extern/pybind11/tests`, which it (obviously) should not pick up.

For now, you can add the tests, and we can add a pytest configuration option to disable searching for tests in `/extern`.

**Comment by LovelyBuggies** [comment]
@henryiii  When I ran the pytest, I found the sequence of operations in`contributing.md` might ambiguous. `python3 -m pytest` occurs after `deactivate`, but the new contributors might not have `pytest`, `pytest-benchmark`,  `pytest-sugar` in their pip root (*note that we only install these packages in `.env`*). I think we should modify the order of operations in the `contributing.md`. 

Rec. I recommend you to move `deactivate` down.
